frontend/src/app/api/SearchOpportunityAPI.ts

```diff
@@ -79,16 +80,26 @@ export default class SearchOpportunityAPI extends BaseApi {
   private buildPagination(
     searchInputs: SearchFetcherProps,
   ): PaginationRequestBody {
-    const { sortby, page } = searchInputs;
+    const { sortby, page, fieldChanged } = searchInputs;
+
+    // When performing an update (query, filter, sortby change) - we want to
+    // start back at the 1st page (we never want to retain the current page).
+    // In addition to this statement - on the client (handleSubmit in useSearchFormState), we
+    // clear the page query param and set the page back to 1.
+    // On initial load (SearchFetcherActionType.InitialLoad) we honor the page the user sent. There is validation guards
+    // in convertSearchParamstoProperTypes keep 1<= page <= max_possible_page
+    const page_offset =
+      searchInputs.actionType === SearchFetcherActionType.Update &&
+      fieldChanged !== "pagination"
+        ? 1
+        : page;
 
-    // TODO: 3/18/24 - API only allows id or number right now
-    // Will need to change these two valid values
     const orderByFieldLookup = {
       opportunityNumber: "opportunity_number",
-      opportunityTitle: "opportunity_number",
-      agency: "opportunity_id",
-      postedDate: "opportunity_id",
-      closeDate: "opportunity_id",
+      opportunityTitle: "opportunity_title",
+      agency: "agency_code",
+      postedDate: "post_date",
+      closeDate: "close_date",
```

Comment:
these were just added to the api recently so we can sortby on them without error